batavia/VirtualMachine.js

```diff
@@ -1417,7 +1421,11 @@ batavia.VirtualMachine.prototype.byte_STORE_MAP = function() {
         case batavia.BATAVIA_MAGIC_35a0:
         case batavia.BATAVIA_MAGIC_34:
             var items = this.popn(3);
-            items[0][items[2]] = items[1];
+            if (items[0].__setitem__) {
+                items[0].__setitem__(items[2], items[1]);
+            } else {
+                items[0][items[2]] = items[1];
+            }
```

Comment:
`List` doesn't have `__setitem__` (yet), and I don't think we use `List` everywhere instead of literal arrays.